README.md

```diff
@@ -105,7 +145,38 @@ httpClient.execute(request: request, deadline: .now() + .milliseconds(1))
 ```
 
 ### Streaming
-When dealing with larger amount of data, it's critical to stream the response body instead of aggregating in-memory. Handling a response stream is done using a delegate protocol. The following example demonstrates how to count the number of bytes in a streaming response body:
+When dealing with larger amount of data, it's critical to stream the response body instead of aggregating in-memory. 
+The following example demonstrates how to count the number of bytes in a streaming response body:
+
+#### Using Swift Concurrency
+```swift
+let httpClient = HTTPClient(eventLoopGroupProvider: .createNew)
+
+let request = HTTPClientRequest(url: "https://apple.com/")
+let response = try await httpClient.execute(request, timeout: .seconds(30))
+print("HTTP head", response)
+
+// if defined, the content-length headers announces the size of the body
+let expectedBytes = response.headers.first(name: "content-length").flatMap(Int.init)
+
+var receivedBytes = 0
+// asynchronously iterates over all body fragments.
+for try await buffer in response.body {
+    // For this example, we are just interested in the size of the fragment
+    receivedBytes += buffer.readableBytes
+    
+    if let expectedBytes = expectedBytes {
+        // if the body size is known, we calculate a progress indicator
+        let progress = Double(receivedBytes)/Double(expectedBytes)
+        print("progress: \(Int(progress * 100))%")
+    }
+    // in case backpressure is needed, all reads will be paused until the end of the for loop.
+}
+print("did receive \(receivedBytes) bytes")
```

Comment:
close AHC?

Reply:
The full example has the shutdown(): https://github.com/swift-server/async-http-client/blob/3405d11bf46e0be3130aa318cceefa1941752618/Examples/StreamingByteCounter/StreamingByteCounter.swift#L25-L52
We would need to nest it in a do catch to be correct but I'm fine with it if you want to add it. I didn't add it because the old one also has no shutdown. However, I'm just discovering that the old examples doesn't create an AsyncHTTPClient, it just assumes that there is one. I see two options:
- remove HTTPClient creation
- or add shutdown
WDYT?

Reply:
I think add the shutdown and nest in a do catch.